cmake/modules/gluecodium/gluecodium/TargetSources.cmake

```diff
@@ -71,8 +76,60 @@ function(apigen_target_sources target)
 
   apigen_list_generated_sources(_generated_files
     ${_source_sets}
+    TARGET ${target}
     GENERATOR "${GENERATOR}"
     BUILD_OUTPUT_DIR "${BUILD_OUTPUT_DIR}")
   source_group("Generated Source Files" FILES ${_generated_files})
-  target_sources(${target} PRIVATE ${_generated_files})
+
+  if(NOT GENERATOR STREQUAL swift)
+    target_sources(${target} PRIVATE ${_generated_files})
+  else()
+    apigen_set_generated_files(${target})
+
+    foreach(_upper_case_source_set ${_source_sets})
+      string(TOLOWER ${_upper_case_source_set} _source_set)
+      # Generated files are marked as such by CMake, but this source file property is on directory scope.
+      # This means for targets in other directories, CMake is not aware that the file is supposed to be
+      # absent during configuration, see issue https://gitlab.kitware.com/cmake/cmake/issues/18399.
+      # Work around this by marking it explicitly as generated and also create dummy files.
+      # When creating these dummy files it is important that not all are already present while building,
+      # otherwise CMake may decide to skip Gluecodium step completely. But this workaround is only
+      # necessary for public/interface sources, so as long as there some private ones without dummy
+      # file, everything is fine.
+      set_property(SOURCE
+            ${BUILD_OUTPUT_DIR}/${APIGEN_GENERATED_cbridge_${_source_set}}
+            ${BUILD_OUTPUT_DIR}/${APIGEN_GENERATED_cbridge_header_${_source_set}}
+            ${BUILD_OUTPUT_DIR}/${APIGEN_GENERATED_cpp_${_source_set}}
+          PROPERTY GENERATED 1)
+
+      foreach(generated_file
+          ${BUILD_OUTPUT_DIR}/${APIGEN_GENERATED_cbridge_${_source_set}}
+          ${BUILD_OUTPUT_DIR}/${APIGEN_GENERATED_cbridge_header_${_source_set}}
+          ${BUILD_OUTPUT_DIR}/${APIGEN_GENERATED_swift_${_source_set}})
+        if(NOT EXISTS "${generated_file}")
+          get_filename_component(directory "${generated_file}" DIRECTORY)
+          file(MAKE_DIRECTORY "${directory}")
+          file(WRITE "${generated_file}" "// Dummy file to be replaced by Gluecodium during build, see also https://gitlab.kitware.com/cmake/cmake/issues/18399")
```

Comment:
If it's dummy file which will be replaced, may then it should contain #error "Dummy file blah blah ..."?

Reply:
Yes, good idea.